### PR TITLE
Snyder l222 land input 2

### PR DESCRIPTION
**Pull request by abigailsnyder**
@bpbond  still didn't do anything with the xml's or logit.type. Just let me know where/what to change for that

#592

### CODE REVIEW

File: R/constants.R [hunk 1]
```diff
@@ -190,6 +190,16 @@ aglu.MIN_SOIL_CARBON_DENSITY <- 0
 aglu.N0_LOGIT_EXP <- 0
 aglu.N0_LOGIT_TYPE <- NA
 
+# fraction of land that is protected
+aglu.PROTECT_LAND_FRACT <- 0.9
+
+# unManaged Land Value
+aglu.UNMANAGED_LAND_VALUE <- 1
```
Comment (bpbond) [inline R/constants.R]: Units?

File: R/zchunk_L222.land_input_2.R [hunk 2]
```diff
@@ -55,182 +86,334 @@ module_aglu_L222.land_input_2_DISABLED <- function(command, ...) {
...
+
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L125.LC_bm2_R_LT_Yh_GLU
+
+
+    # 2. Build tables
+
+    # L222.LN2_Logit: Logit exponent of the second nest
```
Comment (bpbond) [inline R/zchunk_L222.land_input_2.R]: GCAM specialists will understand this comment, but no one else. Expand a bit?

File: R/zchunk_L222.land_input_2.R [hunk 3]
```diff
@@ -55,182 +86,334 @@ module_aglu_L222.land_input_2_DISABLED <- function(command, ...) {
...
+
+    # L222.LN2_MgdCarbon: Carbon content info, managed land in the second nest
+    # Add carbon content info to the master  table to give the
+    # carbon content infor for managed land in the second nest.
+    L222.LC_bm2_R_Mgd2_Yh_GLU %>%
+      filter(year == max(BASE_YEARS)) %>%
+      select(-year, -allocation) %>%
+      left_join_error_no_match(GCAMLandLeaf_CdensityLT, by = c("Land_Type" = "LandLeaf")) %>%
+      rename(Cdensity_LT = Land_Type.y) %>%
+      add_carbon_info(., carbon_info_table = L121.CarbonContent_kgm2_R_LT_GLU) %>%
```
Comment (bpbond) [inline R/zchunk_L222.land_input_2.R]: Why `., `?

**Comment by bpbond** [comment]
>still didn't do anything with the xml's or logit.type. Just let me know where/what to change for that

I still am fuzzy on this. @pralitp can you help/clarify?

**Comment by pralitp** [comment]
Sure. @abigailsnyder, you want me to drop by your office some time to discuss what wasn't clear about it?

**Comment by bpbond** [comment]
If you do can I sit in?!? Feeling like a blockhead w.r.t. this.

**Comment by abigailsnyder** [comment]
@bpbond @pralitp shall I make a quick outlook meeting so that everyone is coordinated?

File: R/constants.R (hunk 1 above)
Comment (abigailsnyder) [inline R/constants.R]: sorry to bother, but @pkyle: I guessed 1975$ per thou km2

Comment (pkyle) [inline R/constants.R]: Yeah that's right, 1975$ per thousand square kilometers. This for land use regions with no commercial uses of land, so no ability to calculate average land value. I think there are 8 or 10 of them, mostly in the Arctic. Anyway it just needs to be a non-zero number so it doesn't blow up the land sharing equations.

**Comment by bpbond** [comment]
Not sure this needs a review from @rynacui , unless you'd particularly like it @abigailsnyder .

**Comment by rynacui** [comment]
I'll take a look later of the batch xml part for learning, but I'll pass for now if you are both fine with it @abigailsnyder @bpbond :)

**Comment by pralitp** [comment]
@rynacui, I walked Abagail and Ben through the XML stuff today and tried to give some context and background for it in hopes that it would be helpful for understanding it.  I would be glad to walk you through as well if you think that would be helpful for you too.

**Comment by rynacui** [comment]
Thanks @pralitp ! I'll let you know!

**Comment by bpbond** [comment]
Done! 🎉

**Pull request merged**